### PR TITLE
Use OpenSSH client instead of a Go implementation

### PR DESCRIPTION
**Pull request by robertodauria**
The PR changes the `forward` command to call the system OpenSSH client instead of relying on a Go implementation.

There are several reasons for this:
- The OpenSSH client seems to be noticeably faster
- Any custom user configuration into `~/.ssh` will be applied. This means we can, for example, easily configure [multiplexing](https://en.wikibooks.org/wiki/OpenSSH/Cookbook/Multiplexing)
- There was an issue I've run into with the Go implementation where if the tunnel was used for a very long time it stopped working and needed to be restarted - not sure what's the reason for that, maybe too many connections?
- The OpenSSH client has been around for a very long time and it's generally much more reliable.

The `forward` function in `forward.go` is left untested as firing up a local SSH server in a Travis build seems to fail consistently - that wasn't the case until some time ago, not sure what changed, but any suggestions are welcome. However, `makeBMCHostname` and `splitPorts` are tested and what's left is the call to `ssh` which is probably correct by inspection.

---
This change is [Reviewable](https://reviewable.io/reviews/m-lab/bmctool/18)

### CODE REVIEW

**Comment by bassosimone** [comment]
> [...] firing up a local SSH server in a Travis build seems to fail consistently [...]

It may be useful to know more about _how_ it fails.

> [...]  any suggestions are welcome

It depends on the error, I guess. However, it's also true that many weird Travis issues can also be bypassed using a Docker container for running tests. So, you may wanna consider that.

File: cmd/forward.go [hunk 1]
```diff
@@ -110,27 +96,21 @@ func forward(dstHost string) {
...
-			Local:  localEndpoint,
-			Server: serverEndpoint,
-			Remote: remoteEndpoint,
-		}
-
-		log.Infof("Forwarding %s -> %s -> %s", localEndpoint, serverEndpoint, remoteEndpoint)
-		errs.Go(tunnel.Start)
+		cmd := exec.Command("ssh", "-N", "-q", tunnelHost, "-L"+
+			strconv.FormatInt(int64(srcPort), 10)+":"+dstHost+":"+
+			strconv.FormatInt(int64(dstPort), 10))
```
Comment (bassosimone) [inline cmd/forward.go]: You may want to generate the string once and use it twice

**Comment by robertodauria** [comment]
> 
> It may be useful to know more about _how_ it fails.
>  
> It depends on the error, I guess. However, it's also true that many weird Travis issues can also be bypassed using a Docker container for running tests. So, you may wanna consider that.

Discussed oob, but I'm leaving the [link](https://travis-ci.com/m-lab/bmctool/builds/119132263) to the failing travis build and a summary here.

I've observed the time it took for the connection to fail, and it looks like a timeout (it's surely more than a minute, while locally it's immediate), but the error message doesn't make much sense to me as it seems to suggest that the channel type (`tcpip-direct`) isn't supported, not that the connection timed out.

I've tried replicating the issue with the most updated `sshserver` package - which I was using to simulate an ssh server - but it still worked perfectly on my machine. That started happening after a completely unrelated change on a different `bmctool` command, but there has been no Travis build for this project for the past month or so.

ACK on the idea of building in a Docker container, but I still think moving to the OpenSSH client brings a lot of advantages on its own for not doing it.

**Comment by bassosimone** [comment]
> ACK on the idea of building in a Docker container, but I still think moving to the OpenSSH client brings a lot of advantages on its own for not doing it.

Sure. I was just suggesting to be lazy, make it work on Docker for you locally and then use Docker also on Travis. As I said, this has served me quite well and surely better than Travis so far.

**Pull request merged**